cloud/deployment/deployment.go

```diff
@@ -415,40 +423,31 @@ func Create(name, workspaceID, description, clusterID, runtimeVersion, dagDeploy
 			if strings.EqualFold(executor, AstroExecutor) || strings.EqualFold(executor, ASTRO) {
 				requestedExecutor = astroplatformcore.CreateDedicatedDeploymentRequestExecutorASTRO
 			}
-			var remoteExecution *astroplatformcore.DeploymentRemoteExecutionRequest
-			if remoteExecutionEnabled {
-				remoteExecution = &astroplatformcore.DeploymentRemoteExecutionRequest{
-					Enabled:                true,
-					AllowedIpAddressRanges: allowedIpAddressRanges,
-					TaskLogBucket:          taskLogBucket,
-					TaskLogUrlPattern:      taskLogURLPattern,
-				}
-			}
 			dedicatedDeploymentRequest := astroplatformcore.CreateDedicatedDeploymentRequest{
-				AstroRuntimeVersion: runtimeVersion,
-				Description:         &description,
-				Name:                name,
-				Executor:            requestedExecutor,
-				IsCicdEnforced:      isCicdEnforced,
-				IsDagDeployEnabled:  dagDeployEnabled,
-				IsHighAvailability:  highAvailabilityValue,
-				IsDevelopmentMode:   &developmentModeValue,
-				ClusterId:           clusterID,
-				WorkspaceId:         workspaceID,
-				Type:                astroplatformcore.CreateDedicatedDeploymentRequestTypeDEDICATED,
-				WorkloadIdentity:    deplWorkloadIdentity,
-				RemoteExecution:     remoteExecution,
+				AstroRuntimeVersion:  runtimeVersion,
+				Description:          &description,
+				Name:                 name,
+				Executor:             requestedExecutor,
+				IsCicdEnforced:       isCicdEnforced,
+				IsDagDeployEnabled:   dagDeployEnabled,
+				IsHighAvailability:   highAvailabilityValue,
+				IsDevelopmentMode:    &developmentModeValue,
+				ClusterId:            clusterID,
+				WorkspaceId:          workspaceID,
+				Type:                 astroplatformcore.CreateDedicatedDeploymentRequestTypeDEDICATED,
+				DefaultTaskPodCpu:    defaultTaskPodCPUPtr,
+				DefaultTaskPodMemory: defaultTaskPodMemoryPtr,
+				ResourceQuotaCpu:     resourceQuotaCPUPtr,
+				ResourceQuotaMemory:  resourceQuotaMemoryPtr,
+				WorkloadIdentity:     deplWorkloadIdentity,
+				RemoteExecution:      remoteExecution,
 			}
-			if dedicatedDeploymentRequest.RemoteExecution == nil || !dedicatedDeploymentRequest.RemoteExecution.Enabled {
-				dedicatedDeploymentRequest.DefaultTaskPodCpu = &defaultTaskPodCpu
-				dedicatedDeploymentRequest.DefaultTaskPodMemory = &defaultTaskPodMemory
-				dedicatedDeploymentRequest.ResourceQuotaCpu = &resourceQuotaCpu
-				dedicatedDeploymentRequest.ResourceQuotaMemory = &resourceQuotaMemory
+			if !remoteExecutionEnabled {
 				if strings.EqualFold(executor, CeleryExecutor) || strings.EqualFold(executor, CELERY) || strings.EqualFold(executor, AstroExecutor) || strings.EqualFold(executor, ASTRO) {
 					dedicatedDeploymentRequest.WorkerQueues = &defautWorkerQueue
 				}
 			}
-			switch schedulerSize {
+			switch schedulerSize { //nolint:dupl
```

Comment:
Nit: if you look at it, the case should be on the dedicated deployment request scheduler size small, and so on for the rest, i.e., dedicated instead of standard. Tiny thing in the existing codebase, but could lead to buggy code someday.

Reply:
Oh yeah, the refs are incorrect. Let's handle that on a different PR.